File: Task-03/Javascript/subtask4.js
const fs = require('fs');

// Read the file
fs.readFile('input.txt', 'utf8', (err, data) => {
    // No explicit error handling
    if (err) throw err;

    const n = data(); 
    let output = '';

    // Generate the top half 
    for (let i = 1; i <= n; i += 2) {
        output += " ".repeat((n - i) / 2) + "*".repeat(i) + '\n';
    }

    // Generate the bottom half 
    for (let i = n - 2; i >= 1; i -= 2) {
        output += " ".repeat((n - i) / 2) + "*".repeat(i) + '\n';
    }

    // Write the result to output.txt
    fs.writeFile('output.txt', output, (err) => {
        // No explicit error handling
        if (err) throw err;

        console.log("Diamond pattern has been ceated and written sucessfully");
    });
});

File: Task-03/Go/subtask4.go
package main

import (
	"io/ioutil"
	"strconv"
	"strings"
)

func main() {
	// Read the input 
	data, _ := ioutil.ReadFile("input.txt")
	input := strings.TrimSpace(string(data))
	n, _ := strconv.Atoi(input)

	var output strings.Builder

	// Upper half of the diamond
	for i := 1; i <= n; i += 2 {
		output.WriteString(strings.Repeat(" ", (n-i)/2))
		output.WriteString(strings.Repeat("*", i))  
		output.WriteString("\n")                         
	}

	// Lower half of the diamond
	for i := n - 2; i > 0; i -= 2 {
		output.WriteString(strings.Repeat(" ", (n-i)/2)) 
		output.WriteString(strings.Repeat("*", i))       
		output.WriteString("\n")                         
	}

	// Write the output to "output.txt"
	ioutil.WriteFile("output.txt", []byte(output.String()), 0644)
}
